kube-client/src/client/body.rs

```diff
@@ -44,6 +44,11 @@ impl Body {
     {
         Body::new(Kind::Wrap(body.map_err(Into::into).boxed_unsync()))
     }
+
+    /// Collect all the pieces of the body
+    pub fn collect(self) -> http_body_util::combinators::Collect<Self> {
```

Comment:
Maybe we want to make this an `async fn` instead of exposing the concrete `Future` impl type?

Reply:
we could await inside the fn and call `to_bytes` on it to avoid the http_body_util type (as done in https://github.com/kube-rs/kube/pull/1445/commits/a5ac263a8fcbe6b97b073bf325f7f2f1eff61b87)

(doing just .await gives you another exposed type [`Collected`](https://docs.rs/http-body-util/latest/http_body_util/struct.Collected.html) so felt better to just collect into what we are most likely to use)